src/Amazon.Common.DotNetCli.Tools/Utilities.cs

```diff
@@ -13,11 +13,37 @@
 using System.Xml.Linq;
 using System.Xml.XPath;
 using Amazon.Util;
+using System.Text.RegularExpressions;
 
 namespace Amazon.Common.DotNetCli.Tools
 {
     public static class Utilities
     {
+        /// <summary>
+        /// Compiled Regex for {VARIABLE} token searches
+        /// </summary>
+        private readonly static Regex EnvironmentVariableTokens = new Regex(@"[$][(].*?[)]", RegexOptions.Compiled);
+
+        /// <summary>
+        /// Replaces {VARIABLE} tokens with environment variables
+        /// </summary>
+        /// <param name="original">original string</param>
+        /// <returns>string with environment variable replacements</returns>
+        public static string ReplaceEnvironmentVariables(string original)
+        {
+            MatchCollection matches = EnvironmentVariableTokens.Matches(original);
+
+            var modified = original;
+
+            foreach (Match m in matches)
+            {
+                var withoutBrackets = m.Value.Substring(2, m.Value.Length - 3);
+                var env = Environment.GetEnvironmentVariable(withoutBrackets);
```

Comment:
I'm thinking if the environment variable is not set and env is either null or "" then the code should not do the replace call. That way if somebody did have a legitimate reason for having a $(somestring) that was not an environment variable it would unlikely be removed.

There should also be a test case for the environment variable not being found.

Reply:
Well, there already was a test for this - but one that verifies that it removed the variable - I know from using a lot of MSBuild and other systems, they always replace $(Identifier) with null / empty string.

**However**, its not a big deal to be cautious especially introducing potentially breaking behavior.   In doing so we now have to get ALL variables and test for the absence of the tokens.  We can no longer just call `Environment.GetEnvironment` since the intended value of the variable could be set to empty.string or null...   

Much to my surprise `System.Environment.GetEnvironmentVariables()`  is seems like a .Net Framework 1.0 API as its not even using a Dictionary<string,string>.  At first wanted to cache all variables using a simple static- but the unit tests fail since the static is initialized only once per app domain / test run.

I would say display a warning for un-replaced variables, but I'll leave that up to you.  Line 48 (the continue statement) of Utilties.cs is where you could add the log statement.